pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -925,10 +925,22 @@ self: super: {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
   };
-  Agda = super.Agda.override {
+  Agda = overrideCabal (super.Agda.override {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
-  };
+  }) (drv: {
+    postInstall = ''
+      # Separate loops to avoid internal error
+      files=($out/share/*-ghc-*/Agda-*/lib/prim/Agda/{Primitive.agda,Builtin/*.agda})
+      for f in "''${files[@]}" ; do
+        $out/bin/agda $f
+      done
+      for f in "''${files[@]}" ; do
+        $out/bin/agda -c --no-main $f
+      done
+      $out/bin/agda-mode compile
+    '';
+  });
 
```

Comment:
Actually, I believe that cabal2nix is the right place for that kind of information. The override files in Nixpkgs are a kludge that shouldn't be used unless we have to. It's much nicer for everyone if cabal2nix generates correct expressions in the first place.